include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -635,6 +720,47 @@ namespace parallel
 
     template <typename Number>
     void
+    Vector<Number>::import(const ReadWriteVector<Number>                  &V,
+                           VectorOperation::values                         operation,
+                           std_cxx11::shared_ptr<const CommunicationPatternBase> communication_pattern)
+    {
+      // If no communication pattern is given, create one. Otherwise, use the
+      // given one.
+      std_cxx11::shared_ptr<const Utilities::MPI::Partitioner> comm_pattern;
+      if (communication_pattern.get() == NULL)
+        {
+          comm_pattern.reset(new Utilities::MPI::Partitioner(locally_owned_elements(),
+                                                             V.get_stored_elements(),
+                                                             get_mpi_communicator()));
+        }
+      else
+        {
+          comm_pattern =
+            std_cxx11::dynamic_pointer_cast<const Utilities::MPI::Partitioner> (communication_pattern);
+          AssertThrow(comm_pattern != NULL,
+                      ExcMessage(std::string("The communication pattern is not of type ") +
+                                 "Utilities::MPI::Partitioner."));
+        }
+      LinearAlgebra::parallel::Vector<Number> tmp_vector(comm_pattern);
+
+      // fill entries from ReadWriteVector into the distributed vector,
+      // including ghost entries. this is not really efficient right now
+      // because indices are translated twice, once for
```

Comment:
I will append that with my next patch when I move the vector to the `ReadWriteVector::distributed` namespace (if that is what comes out). For completeness already now: The two index translations are for nth_index_in_set and once for tmp_vector::operator().